snapcraft/tests/plugins/python/test_python_finder.py

```diff
@@ -0,0 +1,181 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import re
+
+from unittest import mock
+from testtools.matchers import (
+    Equals,
+    MatchesRegex,
+)
+
+from snapcraft.plugins._python import (
+    errors,
+    _python_finder,
+)
+
+from snapcraft import (
+    tests
+)
+
+
+def _create_python_binary(base_dir):
+    python_command_path = os.path.join(
+        base_dir, 'usr', 'bin', 'python-test')
+    os.makedirs(os.path.dirname(python_command_path))
+    open(python_command_path, 'w').close()
+
+
+class GetPythonCommandTestCase(tests.TestCase):
+
+    def test_staged(self):
+        """get_python_command should support staged python"""
+
+        stage_dir = 'stage_dir'
+        install_dir = 'install_dir'
+
+        # Create staged binary
+        _create_python_binary(stage_dir)
+
+        python_command = _python_finder.get_python_command(
+            'python-test', stage_dir, install_dir)
+        self.assertThat(
+            python_command, Equals(os.path.join(
+                stage_dir, 'usr', 'bin', 'python-test')))
+
+    def test_in_part(self):
+        """get_python_command should support in-part python"""
+
+        stage_dir = 'stage_dir'
+        install_dir = 'install_dir'
+
+        # Create installed binary
+        _create_python_binary(install_dir)
+
+        python_command = _python_finder.get_python_command(
+            'python-test', stage_dir, install_dir)
+        self.assertThat(
+            python_command, Equals(os.path.join(
+                install_dir, 'usr', 'bin', 'python-test')))
+
+    def test_staged_and_in_part(self):
+        """get_python_command should prefer staged python over in-part"""
```

Comment:
The why would be good to explain here.

Reply:
I added a blurb to the finder functions instead of the tests.